File: js/empty-commit-2.js
'use strict';

const {BLAKE2b} = require('bcrypto');
const {Tree} = require('urkel');

(async () => {
  const tree = new Tree({
    hash: BLAKE2b,
    bits: 256,
    prefix: './tree'
  });

  await tree.open();

  for (let i = 0; i < 20; i++) {
    const txn2 = tree.txn();
    await txn2.commit();
  }

  // one last empty commit
  const txn = tree.txn();
  await txn.commit();

  await tree.close();
})().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: src/empty-commit-2.c
#include <stdlib.h>
#include <string.h>
#include <assert.h>
#include <stdint.h>
#include <urkel.h>

#include "util.h"

int main() {
  int i;
  urkel_t *db;
  urkel_tx_t *tx;

  db = urkel_open("./tree");
  assert(db != NULL);

  for (i = 0; i < 20; i++) {
    tx = urkel_tx_create(db, NULL);
    assert(tx != NULL);
    assert(urkel_tx_commit(tx));
    urkel_tx_destroy(tx);
  }

  /* one last empty commit. */
  tx = urkel_tx_create(db, NULL);
  assert(tx != NULL);
  assert(urkel_tx_commit(tx));
  urkel_tx_destroy(tx);

  urkel_close(db);
  return 0;
}
